engine/apps/grafana_plugin/views/sync.py

```diff
@@ -49,19 +51,20 @@ def post(self, request: Request) -> Response:
             },
         )
 
-    def get(self, request: Request) -> Response:
+    def get(self, _request: Request) -> Response:
         stack_id = self.instance_context["stack_id"]
         org_id = self.instance_context["org_id"]
-
         token_ok = False
+
         try:
             organization = Organization.objects.get(stack_id=stack_id, org_id=org_id)
-            if organization.api_token_status == Organization.API_TOKEN_STATUS_PENDING:
-                return Response(status=status.HTTP_202_ACCEPTED)
```

Comment:
Looking at this I see why it was collapsed but I wonder if we should instead handle all 3 cases both here and the plugin.  The plugin should continue polling on 202 instead of checking if the token_ok field is present.  Here we should return an error code here if token_status == FAILED.  As it stands this get request will give us the same answer for PENDING and FAILED.

Reply:
fair point, I'll add this condition back 👍 

Currently though, the UX is the same, whether or not this condition exists. I think the only benefit of updating the UI to poll based on HTTP 202, is that if the sync fails (and `token_ok` is False) then the sync polling would exit earlier, rather than looping for the hardcoded amount of time. Am I stating that correctly?

if you think it's worthwhile implementing, I propose we create a separate issue and address it there.